### ISSUE TITLE
convert_subnet_to_litgpt is broken for QK-norm and Gemma models

### ISSUE THREAD

**Issue by Neonkraft** [whittle-org/whittle#366]
**Describe the bug**
`whittle.convert.copy_weights_to_litgpt` / `convert_subnet_to_litgpt` produces a LitGPT model whose forward output does not match the corresponding whittle sub-network. Four distinct issues cause this:

1. `whittle.modules.rmsnorm.RMSNorm` did not implement the `extract_weights` method (it was misspelled as `extract_weight`)
2. QK-normalization weights (`attn.norm_q`, `attn.norm_k`) are never copied to the converted LitGPT model, so any model with `config.norm_qk=True` (e.g. OLMo-2, Qwen3, Gemma-3) runs with freshly-initialized QK-norms on the LitGPT side.
3. For models with `add_unit_offset=True` RMSNorm (Gemma family), a double unit-offset is applied. Whittle's `RMSNorm.extract_weights()` already folds `(1 + w)` into the returned weight, and that value is written into LitGPT's RMSNorm, which then re-applies `(1 + weight)` in forward. Net effect: `(2 + w) * x_normed` instead of `(1 + w) * x_normed`.
4. For models with `config.attention_scores_scalar` set (Gemma family), `create_litgpt_config_for_subnet` leaves the supernet's scaler in the converted config. Whittle's attention, however, recomputes `sub_attention_scaler = n_embd // n_head` for the sub-network. The converted LitGPT therefore scales attention logits by `1/sqrt(supernet_scalar)` while the whittle sub-network uses `1/sqrt(n_embd // n_head)`.

**To Reproduce**

1. Install whittle from `main` (pre-fix).
2. Rename `whittle.modules.rmsnorm.RMSNorm.extract_weight` to `whittle.modules.rmsnorm.RMSNorm.extract_weights`
3. Run:
```
import torch
from litgpt import Config
from whittle.convert import convert_subnet_to_litgpt
from whittle.models.gpt import GPT

config = Config.from_name(
    "gemma-2-9b", n_layer=2, n_embd=32, intermediate_size=86, padded_vocab_size=10000
)
config.fix_head_size = True
whittle_model = GPT(config).eval()
subnet = dict(
    sub_network_n_embd=16, sub_network_intermediate_size=43,
    sub_network_num_heads=8, sub_network_n_layers=1,
    sub_network_query_groups=4, sub_network_head_size=config.head_size,
)
lit = convert_subnet_to_litgpt(whittle_model, subnet).eval()
whittle_model.set_sub_network(**subnet)

x = torch.tensor([[9856, 23, 491, 1536, 304]], dtype=torch.int32)
out_whittle = whittle_model(x)
out_lit = lit(x)
print((out_whittle - out_lit).abs().max())   # ~9.5e-01
```
3. Outputs differ by ~1e-1 to ~2e0; `torch.allclose(..., atol=1e-3)` fails for Gemma models. For OLMo-2 / Qwen3 the QK-norm copy is missing and outputs silently diverge whenever those layers are non-identity.
**Expected behavior**
The converted LitGPT model should reproduce the whittle sub-network output bitwise (i.e. `torch.allclose(out_whittle, out_lit, atol=1e-6)` passes) for all supported model families.

**Screenshots**
N/A — numerical output only.

**Environment and Installation:**

- OS: Ubuntu 22.04.5
- Environment: conda
- Python version: 3.11.13
- whittle version / commit hash: `main` @ `4101722` ("fix: fix qk normalization in attention layers (#365)")
